test.sh

```diff
@@ -139,116 +144,116 @@ GOBIN=${GOBIN:-$HOME/gopath/bin}
 #
 # Run Go Vet, a correctness-focused static analysis tool
 #
-
-start_context "test/vet"
-run_and_comment go vet ./...
-end_context #test/vet
+if [[ "$RUN" =~ "vet" ]] ; then
+  start_context "test/vet"
+  run_and_comment go vet ./...
+  end_context #test/vet
+fi
 
 #
 # Run Go Lint, a style-focused static analysis tool
 #
-
-start_context "test/golint"
-[ -x "$(which golint)" ] && run golint ./...
-end_context #test/golint
+if [[ "$RUN" =~ "lint" ]] ; then
+  start_context "test/golint"
+  [ -x "$(which golint)" ] && run golint ./...
+  end_context #test/golint
+fi
 
 #
 # Ensure all files are formatted per the `go fmt` tool
 #
-start_context "test/gofmt"
-check_gofmt() {
-  unformatted=$(find . -name "*.go" -not -path "./Godeps/*" -print | xargs -n1 gofmt -l)
-  if [ "x${unformatted}" == "x" ] ; then
-    return 0
-  else
-    V="Unformatted files found.
-    Please run 'go fmt' on each of these files and amend your commit to continue."
-
-    for f in ${unformatted}; do
-      V=$(printf "%s\n - %s" "${V}" "${f}")
-    done
-
-    # Print to stdout
-    printf "%s\n\n" "${V}"
-    [ "${TRAVIS}" == "true" ] || exit 1 # Stop here if running locally
-    return 1
-  fi
-}
+if [[ "$RUN" =~ "fmt" ]] ; then
+  start_context "test/gofmt"
+  check_gofmt() {
+    unformatted=$(find . -name "*.go" -not -path "./Godeps/*" -print | xargs -n1 gofmt -l)
+    if [ "x${unformatted}" == "x" ] ; then
+      return 0
+    else
+      V="Unformatted files found.
+      Please run 'go fmt' on each of these files and amend your commit to continue."
+
+      for f in ${unformatted}; do
+        V=$(printf "%s\n - %s" "${V}" "${f}")
+      done
+
+      # Print to stdout
+      printf "%s\n\n" "${V}"
+      [ "${TRAVIS}" == "true" ] || exit 1 # Stop here if running locally
+      return 1
+    fi
+  }
 
-run_and_comment check_gofmt
-end_context #test/gofmt
+  run_and_comment check_gofmt
```

Comment:
These tooling checks don't set FAILURE and so, if they come in from a non-boulder team member's repo, they "pass" because we don't call the github-pr-status command.

I think it's time to fix that, maybe.